### PR TITLE
Extract py runtime

### PR DESCRIPTION
**Pull request by rprimet**
Open draft PR for discussion, following #245

### CODE REVIEW

**Comment by rprimet** [comment]
@denismerigoux I believe the remaining CI error comes from the fact that the python venv may be persistent across CI runs?

**Comment by denismerigoux** [comment]
Yes the Python `venv` is persistent across the short CI runs that checks PR validity. The command for these runs is in https://github.com/CatalaLang/catala/blob/master/.github/workflows/run-make-all.yml. I noticed that this PR changes something concerning the Python venv; what should I do to restore the CI ?

**Comment by rprimet** [comment]
Since the env setup step is suite short, maybe rm'ing the `env` directory and running `setup_env.sh` before each run of `make all` would be a solution ? (Even without deleting the env directory on each run, we should run `pip install . ` in the python runtime directory before each run if we go forward with an installable runtime -- which `setup_env.sh` does)

**Comment by denismerigoux** [comment]
I added you as an admin of the Inria CI project (ci.inria.fr) where the machines are hosted. You can login with user `ci` and a password I'll send you via email. The Catala repo where the actions happen is in `/builds/actions-runner/catala/catala` on each one of the two machines and `tmux a` brings you to a session where the GitHub actions runner listents to income job requests.

File: french_law/python/src/allocations_familiales.py [hunk 1]
```diff
@@ -1,59 +1,58 @@
 # This file has been generated by the Catala compiler, do not edit!
```
Comment (denismerigoux) [inline french_law/python/src/allocations_familiales.py:1]: Did you format this with `autopep8` ?

File: runtimes/python/catala/src/catala/runtime.py [hunk 2]
```diff
@@ -70,6 +70,9 @@ def __eq__(self, other: object) -> bool:
 
     def __str__(self) -> str:
         return self.value.__str__()
+    
+    def __repr__(self) -> str:
```
Comment (denismerigoux) [inline runtimes/python/catala/src/catala/runtime.py:74]: Thanks!

File: runtimes/python/catala/src/catala/runtime.py [hunk 3]
```diff
@@ -70,6 +70,9 @@ def __eq__(self, other: object) -> bool:
```
Comment (denismerigoux) [inline runtimes/python/catala/src/catala/runtime.py:70]: Good idea to put the runtimes in a separate folder. I should do the same for OCaml.

**Review by denismerigoux — APPROVED** [review]
Thanks very much @rprimet!

File: french_law/python/src/allocations_familiales.py (hunk 1 above)
Comment (rprimet) [reply french_law/python/src/allocations_familiales.py:1]: Nope, I just did, thanks!

(BTW, any reason for choosing autopep8 over black?)

Comment (denismerigoux) [reply french_law/python/src/allocations_familiales.py:1]: I didn't know black :)

**Review by denismerigoux — APPROVED** [review]

**Pull request merged**